### PR TITLE
[Foundation overlay] Enable custom AnyHashable representation for NSMeasurement

### PR DESCRIPTION
**Pull request by DougGregor**
Enable custom AnyHashable representation for NSMeasurement

Resolves [rdar://problem/27539951](rdar://problem/27539951).

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@gribozavr , do you mind taking a look?

File: validation-test/stdlib/AnyHashable.swift.gyb [hunk 1]
```diff
@@ -870,6 +870,15 @@ AnyHashableTests.test("AnyHashable(NSError).base") {
...
   expectTrue(type(of: ah.base) is NSError.Type)
 }
+
+if #available(OSX 10.12, iOS 10.0, tvOS 10.0, watchOS 3.0, *) {
+  AnyHashableTests.test("Measurement.base") {
+    let measurement = NSMeasurement(doubleValue: 100,
+                                    unit: UnitLength.kilometers)
+    let ah = AnyHashable(measurement)
+    expectTrue(type(of: ah.base) is Measurement.Type)
+  }
```
Comment (gribozavr) [inline validation-test/stdlib/AnyHashable.swift.gyb]: Please move the tests to `test/stdlib/TestMeasurement.swift` instead.  Please follow the pattern in `test_AnyHashableCreatedFromNSCalendar` from `test/stdlib/TestCalendar.swift`.

**Review by gribozavr — CHANGES_REQUESTED** [review]

File: validation-test/stdlib/AnyHashable.swift.gyb (hunk 1 above)
Comment (DougGregor) [inline validation-test/stdlib/AnyHashable.swift.gyb]: Okay, thanks!

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@gribozavr updated, thank you!

File: test/stdlib/TestMeasurement.swift [hunk 2]
```diff
@@ -147,6 +147,36 @@ class TestMeasurement : TestMeasurementSuper {
...
+
+    func test_AnyHashableCreatedFromNSMeasurement() {
+        if #available(iOS 8.0, *) {
+            let values: [NSMeasurement] = [
+                NSMeasurement(doubleValue: 100, unit: UnitLength.meters),
+                NSMeasurement(doubleValue: 100, unit: UnitLength.kilometers),
+                NSMeasurement(doubleValue: 100, unit: UnitLength.kilometers),
+            ]
+            let anyHashables = values.map(AnyHashable.init)
+            expectEqual(Measurement.self, type(of: anyHashables[0].base))
```
Comment (gribozavr) [inline test/stdlib/TestMeasurement.swift]: LGTM with an explicit generic type parameter on `Measurement` in the assertion.

**Review by gribozavr — APPROVED** [review]

File: test/stdlib/TestMeasurement.swift [hunk 3]
```diff
@@ -147,6 +147,36 @@ class TestMeasurement : TestMeasurementSuper {
...
+        let anyHashables = values.map(AnyHashable.init)
+        expectEqual(Measurement<UnitLength>.self, type(of: anyHashables[0].base))
+        expectEqual(Measurement<UnitLength>.self, type(of: anyHashables[1].base))
+        expectEqual(Measurement<UnitLength>.self, type(of: anyHashables[2].base))
+        expectNotEqual(anyHashables[0], anyHashables[1])
+        expectEqual(anyHashables[1], anyHashables[2])
+    }
+
+    func test_AnyHashableCreatedFromNSMeasurement() {
+        if #available(iOS 8.0, *) {
```
Comment (parkera) [inline test/stdlib/TestMeasurement.swift]: This availability check doesn't make sense to me - NSMeasurement is new in iOS 10 and OS X 10.12.

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: test/stdlib/TestMeasurement.swift (hunk 3 above)
Comment (DougGregor) [inline test/stdlib/TestMeasurement.swift]: Good catch; this whole test is already iOS 10/macOS 10.12. Fixed in a subsequent commit.

File: test/stdlib/TestMeasurement.swift (hunk 2 above)
Comment (DougGregor) [inline test/stdlib/TestMeasurement.swift]: Will do, thanks!

**Pull request merged**